### PR TITLE
Parser

### PR DESCRIPTION
**Pull request by meghasinghania22**
Once the type checking is implemented, I will move some of the checks from the Parser to Typechecker

### CODE REVIEW

File: frontend/src/main/input.tdot [hunk 1]
```diff
@@ -1,54 +1,36 @@
...
-	"eating out": { budget: 200, track: true },
-	coffee: { budget: 80, track: true },
-	transport,
-	groceries: { track: true },
-	bills,
-	other
-  }
-}
+create sheet monthly_budget
+  add date September 2019,
```
Comment (ayapcck) [inline frontend/src/main/input.tdot:2]: doesn't this violate the single lookahead thing? And also what was last on the grammar in slack? The "create" token was for a spreadsheet, then "add" was for a sheet only

**Review by sor2 — DISMISSED** [review]

File: frontend/src/main/input.tdot (hunk 1 above)
Comment (ayapcck) [reply frontend/src/main/input.tdot:2]: I also don't see this being wrapped in anything that signifies it creates a spreadsheet document. i.e. including "create SHEET_NAME sheet+"

File: frontend/src/main/input.tdot [hunk 2]
```diff
@@ -1,4 +1,4 @@
-create sheet monthly_budget
+create Budget sheet monthly_budget
```
Comment (ayapcck) [inline frontend/src/main/input.tdot:1]: The SSTitle is for the spreadsheet and not an individual sheet. The names for the sheets are created based on the type.

**Review by atsang36 — APPROVED** [review]

File: frontend/src/main/input.tdot (hunk 2 above)
Comment (meghasinghania22) [reply frontend/src/main/input.tdot:1]: I created SSTitle keeping in mind the Sheet name tbh. I can work on adding a new node which takes in the Spreadsheet name later tonight

Comment (ayapcck) [reply frontend/src/main/input.tdot:1]: Well in my PR I've reworked it so SSTitle corresponds to SpreadSheetTitle, as we don't need the nodes to know their name, only the type. The name is inferred from type. So you don't have to worry about reworking things

**Review by ayapcck — APPROVED** [review]

**Review by sor2 — APPROVED** [review]

**Pull request merged**